src/Microsoft.ML.Tokenizers/Model/WordPieceTokenizer.cs

```diff
@@ -0,0 +1,858 @@
+                            IReadOnlyDictionary<string, int>? specialTokens,
+                            string unknownToken,
+                            string continuingSubwordPrefix,
+                            int maxInputCharsPerWord,
+                            bool disposeStream)
+        {
+            if (vocabStream is null)
+            {
+                throw new ArgumentNullException(nameof(vocabStream));
+            }
+
+            try
+            {
+                (Dictionary<StringSpanOrdinalKey, int> vocab, Dictionary<int, string> vocabReverse) = LoadVocabAsync(vocabStream, useAsync: false).GetAwaiter().GetResult();
+
+                return new WordPieceTokenizer(vocab, vocabReverse, preTokenizer, normalizer, specialTokens, unknownToken, continuingSubwordPrefix, maxInputCharsPerWord);
+            }
+            finally
+            {
+                if (disposeStream)
+                {
+                    vocabStream.Dispose();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Create a new instance of the <see cref="WordPieceTokenizer"/> class asynchronously.
+        /// </summary>
+        /// <param name="vocabFilePath">The path to the WordPiece vocab file.</param>
+        /// <param name="preTokenizer">The PreTokenizer to use.</param>
+        /// <param name="normalizer">The Normalizer to use.</param>
+        /// <param name="specialTokens">The dictionary containing the special tokens and their corresponding ids.</param>
+        /// <param name="unknownToken">The unknown token. A token that is not in the vocabulary cannot be converted to an ID and is set to be this token instead.</param>
+        /// <param name="continuingSubwordPrefix">The prefix to use for sub-words that are not the first part of a word.</param>
+        /// <param name="maxInputCharsPerWord">The maximum number of characters to authorize in a single word.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A new instance of the <see cref="WordPieceTokenizer"/> class.</returns>
+        /// <remarks>
+        /// If the <paramref name="preTokenizer"/> is null, the whitespace pre-tokenizer will be used.
+        /// </remarks>
+        public static async Task<WordPieceTokenizer> CreateAsync(
+                                string vocabFilePath,
+                                PreTokenizer? preTokenizer = null,
+                                Normalizer? normalizer = null,
+                                IReadOnlyDictionary<string, int>? specialTokens = null,
+                                string unknownToken = "[UNK]",
+                                string continuingSubwordPrefix = DefaultContinuingSubwordPrefix,
+                                int maxInputCharsPerWord = DefaultMaxInputCharsPerWord,
+                                CancellationToken cancellationToken = default) =>
+            await CreateAsync(
```

Comment:
This should either remove the async/await or it should add ConfigureAwait. Same with some of the other one-liner delegations.